src/boot/efi/boot.c

```diff
@@ -2428,6 +2600,11 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
                         continue;
                 }
 
+                if (entry->type == LOADER_SIGNATURE_KEYS) {
```

Comment:
You're actually re-inventing entry->call here, which is supposed to be for this kind of thing.

Reply:
yes, please just install a "call" function in the ConfigEntry instead of this if block

Reply:
Just read all the key data into ConfigEntry during discovery. That way you ensure that you have everything you need to enroll a selected key. Meanwhile ConfigEntry->call would only need the ConfigEntry itself as parameter.

Reply:
hmm, you mean load the key data? i am not totally sure. the files might be large (db/dbx can be quite something, no?) might be slow, risky and use too much memory? dunno, i'd be defensive there, and load the actual key data only when we actually are about to go for it.

Reply:
Since they have to fit into NVRAM they can't be gigantic, any computer running an EFI will be able to handle that. And reading them all in also allows for not even showing entries that are incomplete anyways.

Reply:
I originally was leaning towards @poettering opinion. Even though I agree that most files are probably not huge.

The other problem is that the call function does not have any context as of now as it is only used for a single purpose. I would need to pass it some context (probably the whole Config object and the Entry that triggered it). Would that be ok?

If it sounds good I can propose an implementation for that probably tomorrow.

Reply:
Of course adjusting the params on it is fine, it's not like we have some public api in a bootloader to uphold.